include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
+
+#include <stdint.h>
+
+#undef min
+#undef max
+#undef DeleteFile
+#undef GetCurrentTime
+
+#ifndef strcasecmp
+#define strcasecmp _stricmp
+#endif
+
+#ifndef snprintf
+#define snprintf _snprintf
+#endif
+
+#ifndef ROCKSDB_PRIszt
+#define ROCKSDB_PRIszt "Iu"
+#endif
```

Comment:
Logv I already moved to env.cc, however, there are many places in env.h that depend on the public_port.h to build

From: Siying Dong [mailto:notifications@github.com]
Sent: Monday, July 13, 2015 6:02 PM
To: facebook/rocksdb
Cc: Dmitri Smirnov
Subject: Re: [rocksdb] Ensure Windows build w/o port/port.h in public headers (#652)


In include/rocksdb/public_port.h<https://github.com/facebook/rocksdb/pull/652#discussion_r34529150>:

> +#undef min

> +#undef max

> +#undef DeleteFile

> +#undef GetCurrentTime

> +

> +#ifndef strcasecmp

> +#define strcasecmp _stricmp

> +#endif

> +

> +#ifndef snprintf

> +#define snprintf _snprintf

> +#endif

> +

> +#ifndef ROCKSDB_PRIszt

> +#define ROCKSDB_PRIszt "Iu"

> +#endif

I only find Logger::Logv, which should be able to move to util/env.cc. Any other places I missed?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/rocksdb/pull/652/files#r34529150>.